File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/converters/EmploymentConverter.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.converters;

import androidx.room.TypeConverter;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.response.Employment;

public class EmploymentConverter {
    @TypeConverter
    public static String fromEmployment(Employment employment){
        return employment.getTitle();
    }

    @TypeConverter
    public static Employment toEmployment(String employment){
        return new Employment(employment);
    }

}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/converters/EmploymentConverter.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.converters

import androidx.room.TypeConverter
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.reponse.Employment

object EmploymentConverter {

    @TypeConverter
    @JvmStatic
    fun fromEmployment (employment: Employment): String{
        return employment.title
    }

    @TypeConverter
    @JvmStatic
    fun toEmployment(employment: String): Employment {

        return Employment(employment, employment)
    }
}